invoice/invoice.go

```diff
@@ -0,0 +1,966 @@
+package invoice
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"strconv"
+
+	// TODO: remove these deps?
+	"github.com/btcsuite/btcd/chaincfg"
```

Comment:
Had some problems trying to do this. I must update the version of `roasbeef/btcd` found in `glide.yaml`, but this fails since `neutrino` relies on the same old version. Must I first update `neutrino` to use the new verison, then update `lnd` to point to the new `neutrino` and `btcd`?

Reply:
fixed by pulling latest master 👍